scvi/module/_jaxvae.py

```diff
@@ -60,29 +65,34 @@ class FlaxDecoder(nn.Module):
     n_hidden: int
     is_training: Optional[bool] = None
 
-    @nn.compact
+    def setup(self):
+        self.dense1 = Dense(self.n_hidden)
+        self.dense2 = Dense(self.n_hidden)
+        self.dense3 = Dense(self.n_hidden)
+        self.dense4 = Dense(self.n_input)
+
     def __call__(self, z: jnp.ndarray, batch: jnp.ndarray, is_training: bool):
         disp = self.param(
             "disp", lambda rng, shape: jax.random.normal(rng, shape), (self.n_input, 1)
         )
         is_training = nn.merge_param("is_training", self.is_training, is_training)
```

Comment:
we can get rid of these lines and move all the layers into setup (including batch norm and dropout).